### PR TITLE
add a new Hai Hand to hand weapon 

### PR DESCRIPTION
**Pull request by Brick63**
a little While ago @tehhowch asked me on the discord to create a H2H weapon for the Hai and this is what i came up with. 
From what i could gather this is aimed at making it harder to farm the unfettered ships
i feel this would add more of a sense of escalation to the game and add to it overall

here are the rifle's assets:
[hai rifle assets.zip](https://github.com/endless-sky/endless-sky/files/1622953/hai.rifle.assets.zip)

### CODE REVIEW

**Comment by mdsmestad** [comment]
So you might call this...a squirrel gun! Haha I'm so sorry

File: data/hai outfits.txt [hunk 1]
```diff
@@ -600,3 +600,12 @@ outfit `"Bufaer" Atomic Steering`
...
 	description "Nine out of ten human captains interviewed in Hai space say that they would choose Hai atomic engines over those of Deep Sky any day... that is, if they could afford such expensive engines."
+
+outfit "Pulse Rifle"
+	category "Hand to Hand"
+	cost 11000
+	thumbnail "outfit/hai rifle"
+	"capture attack" .8
+	"capture defense" 1.3
+	"unplunderable" 1
+	description "The Pulse Rifle was first developed by the Hai to help protect their ships from boarders during the Korath raids, and while its since been adapted to not be lethal outright if enough bolts hit in a short period of time the nerves will fry, shutting down the body and killing the target."
```
Comment (KiLEdEnNis) [inline data/hai outfits.txt]: first of all, this single sentence is way too long and confusing
and "and while its since been adapted to not be lethal outright if enough bolts hit in a short period of time the nerves will fry" sounds incorrect and probably is

maybe try
"The Pulse Rifle was first developed by the Hai to help protect their ships from boarders during the Korath raids. While it has since been adapted to not be lethal outright, if several bolts hit a target rapidly, it will shut down the nervous system, killing the target."

Comment (Brick63) [reply data/hai outfits.txt]: yeah thanks im not the best at sentence setup

File: data/hai ships.txt [hunk 2]
```diff
@@ -36,6 +36,7 @@ ship "Aphid"
 		"Hai Fissure Batteries"
 		"Hai Corundum Regenerator"
 		"Hai Williwaw Cooling"
+		"Pulse Rifle"
```
Comment (tehhowch) [inline data/hai ships.txt]: Drawing on feedback given for #3427, the variants of a given model should have different numbers of rifles.

I think more aggressive variants (e.g. one with an offensive turret vs defensive outfits only) should probably have more rifles than their counterparts, since these ships are more likely to go on the attack and become disabled vs those that are conservative and in a supporting role.

(i.e. "Aphid" -> 2 or 3 Pulse Rifles, while "Aphid (Anti-Misisle)" staying at 1 makes sense)

Comment (Brick63) [reply data/hai ships.txt]: okay I will work on that

**Comment by MCOfficer** [comment]
since most of hai space is completely free of threads, i can't see merchants outfitting their freighter with these. it feels more like something all military ships would do

**Comment by Brick63** [comment]
its only on hai ships atm as those are tho most likely to be on the front lines against the unfettered

File: data/hai outfits.txt [hunk 3]
```diff
@@ -600,3 +600,12 @@ outfit `"Bufaer" Atomic Steering`
...
 	description "Nine out of ten human captains interviewed in Hai space say that they would choose Hai atomic engines over those of Deep Sky any day... that is, if they could afford such expensive engines."
+
+outfit "Pulse Rifle"
+	category "Hand to Hand"
+	cost 11000
+	thumbnail "outfit/hai rifle"
+	"capture attack" .8
+	"capture defense" 1.3
+	"unplunderable" 1
+	description "The Pulse Rifle was first developed by the Hai to help protect their ships from boarders during the Korath raids. While it has since been adapted to not be lethal outright, if several bolts hit a target rapidly, it will shut down the nervous system, killing the target."
```
Comment (tehhowch) [inline data/hai outfits.txt]: Maybe reword to  `"The Pulse Rifle was first developed by the Hai to help protect their ships from boarders during the Korath raids. While it has since been adapted to more effectively incapacitate its target, if several bolts land in quick succession, the accumulated charge has a lethal effect on most beings' nervous systems."`

Also, "Pulse Rifle" or "Ion Pulse Rifle"?

Comment (Brick63) [reply data/hai outfits.txt]: its the pulse rifle since its just been adjusted the original one was made to be lethal and has since been adjusted to the more passive weapon you see now

File: data/hai outfits.txt [hunk 4]
```diff
@@ -608,4 +608,4 @@ outfit "Pulse Rifle"
 	"capture attack" .8
 	"capture defense" 1.3
 	"unplunderable" 1
-	description "The Pulse Rifle was first developed by the Hai to help protect their ships from boarders during the Korath raids. While it has since been adapted to not be lethal outright, if several bolts hit a target rapidly, it will shut down the nervous system, killing the target."
+	description "The Pulse Rifle was first developed by the Hai to help protect their ships from boarders during the Korath raids. While it has since been adapted to more effectively incapacitate its target, if several bolts land in quick succession, the accumulated charge has a lethal effect on most beings' nervous systems."
```
Comment (Bladewood) [inline data/hai outfits.txt]: `beings` doesn't sound right. Do what the biodefenses did and use `species` instead.

Comment (Brick63) [reply data/hai outfits.txt]: that does sound better

**Comment by Nescio0** [comment]
On the one hand, I like the new image. On the other, I don't think introducing it would be a good idea.

Making it available would render human laser rifles immediately obsolete. 

Furthermore, Hai space is rather peaceful; there are no pirates, marauders, Korath, or other raiders. The only fighting is the ongoing conflict with the Unfettered in and around Wah Ki. However, they typically blow up each other's ships, they never board or attempt to capture. According to the game's lore, Hai highly value life and avoid risking it unnecessarily. So why would they want to have hand to hand guns?

**Comment by Amazinite** [comment]
>Hai space is rather peaceful; there are no pirates, marauders, Korath, or other raiders. The only fighting is the ongoing conflict with the Unfettered in and around Wah Ki. 

Since the main reason behind this seems to be making Unfettered ships harder to cap, I'd point to what Tehhowch said:
>It would be nice to give the Unfettered more rifles on average than their friendly counterparts, but as both sects use the same variants that's only possible if we create ship variants exclusively for the Unfettered, which is not really necessary given that the two share outfits and in large part, ships.

and *actually* split Unfettered ships into their own variants, and only Unfettered ships would have Pulse Rifles, and only Unfettered worlds could sell this weapon.

>Making it available would render human laser rifles immediately obsolete.

Not necessarily. Pulse RIfles make Laser Rifles as obsolete as Frag Grenades do; it's all a matter of what the player is able to purchase. In the end, the strongest H2H weapon makes all other H2H weapons obsolete anyways, and the Pulse RIfle isn't anywhere close to the strongest.

**Comment by Brick63** [comment]
Nescio0 the weapon says that it's non lethal however enough hits will kill something so it's more of a morality issue

**Comment by Nescio0** [comment]
> actually split Unfettered ships into their own variants, and only Unfettered ships would have Pulse Rifles, and only Unfettered worlds could sell this weapon.

The proposal as it is currently adds these weapons to all Hai ships and all Hai outfitters. And restricting it to be for sale at Unfettered outfitters only won't make a real difference, since those “Deliver Aid” missions allow you to land there for free, allowing you to purchase as many of them as you like. 
Although I'm not opposed to making Unfettered ships harder to seize, I doubt the current proposal would be the perfect solution.

>  Pulse RIfles make Laser Rifles as obsolete as Frag Grenades do

Grenades do not make rifles obsolete. Grenades (1.3/0.3) have a higher attack bonus, rifles (0.6/0.8) a better defence, so you'd need both, at least until you've seized a Korath Raider. These pulse rifles (0.8/1.3) are significantly stronger than laser rifles and only slightly more expensive. The Hai have no trade restrictions, therefore these weapons are immediately available; why would anybody choose human rifles if you can also buy clearly superior Hai rifles?

> Nescio0 the weapon says that it's non lethal however enough hits will kill something so it's more of a morality issue

Really?

**Comment by Brick63** [comment]
Read the description

**Comment by Nescio0** [comment]
“This weapon is not lethal unless you decide to use it to kill.” 

Doesn't this apply to all weapons? And does such a description makes it suddenly morally sound?

**Comment by Amazinite** [comment]
>Grenades do not make rifles obsolete. Grenades (1.3/0.3) have a higher attack bonus, rifles (0.6/0.8) a better defence, so you'd need both,

Defense really isn't important to the player at all. Maybe when attacking a World-Ship, but that's a single extreme case where you're going to be using the best stuff you can get anyway, in which case the Laser Rifle is obsolete to Nerve Gas + Biodefenses.

**Comment by Brick63** [comment]
Not entirely, the projectiles shot have small electrical charges a few are non lethal while if enough hit before the target has recovered from the shocks it can lead to nerve damage and possibly death

**Comment by Brick63** [comment]
Also the KRR is no where close to non lethal

**Comment by Brick63** [comment]
(Gosh why do I not say everything at once)
The pulse rifle is a non-lethal weapon that can be lethal if used that way, it's not fully non-lethal since it may need to be used to stop raids by races like the korath as was it's original purpose but since those stopped it's become more of a policing tool versus the unfettered thus it's adjustment to being non-lethal

**Comment by Nescio0** [comment]
> Defense really isn't important to the player at all. Maybe when attacking a World-Ship, but that's a single extreme case where you're going to be using the best stuff you can get anyway, in which case the Laser Rifle is obsolete to Nerve Gas + Biodefenses.

It is to me. As far as I know, only your flagship can board; your other ships can be seized though, and therefore are in need of defensive rather than offensive weapons. I am not interested in seizing world-ships, however, a fully crewed Bactrian with 245 rilfes *and* 245 grenades is a more likely to capture a Korath *Raider* than a Bactrian with either grenades or rifles, because the attacked can also fight back and decide to attempt to seize your ship.
By the way, where can one get biodefences?

**Comment by Amazinite** [comment]
>By the way, where can one get biodefences?

You have to capture Pug ships to get the biodefenses from them.

**Comment by Nescio0** [comment]
> You have to capture Pug ships to get the biodefenses from them.

Well, I never did; except for a brief interlude in the campaign, the Pug are friendly. I might as well attack the Quarg.

Anyway, Pug and Korath weapons are not for sale, therefore it is not a problem they are superior to anything human. However, the proposed Hai weapon can be immediately purchased and is significantly better, which I think is problematic.

**Comment by jafdy** [comment]
If you are attacking then the defense value is useless. So the rifles are only good if you need to defend yourself for some reason, most likely that you have been really stupid in your attack.

**Comment by Brick63** [comment]
this outfit in effect is to make it more costly to capture hai ships as it would be weird to have the hai using human tech

**Comment by ghoulavenger** [comment]
While I'm perfectly fine with Hai having their own H2H tech, I don't think it would be that weird for the Hai to be using human tech at all. They have human captains there lorewise, and they're likely to be competent ones if they're willing to go through the wormhole (which means they more than likely have access to rifles and grenades but maybe not the illegal nerve gas). 

The problem is that the Hai are a peaceful lot, which means that most of these weapons are probably in the hands of the Unfettered (which are unlikely to have good relations with human captains) or Hai assault ships (which will probably be issued specific weapons for the task and without official trade agreements likely local in origin) to help deal with the occasional Unfettered raid. I can't see the typical Hai captain using any H2H weapons at all. 

In any case I like the rifle, it looks cool. But I'm not really big on arming Hai ships, Unfettered (more permissible) or not. Arming ships that are supposed to be hard to capture, fine, but just doing it in general I think is just a nerf to capturing which I don't feel needs it.

**Comment by Tadrix** [comment]
I, for one, don't see any problems with this new weapon.

1. **We don't need to make separate ship variants for the Unfettered.** Seeing as this Squirrel Gun has a less lethal setting, the peaceful Hai would most likely carry it for all sorts of security related activities. Sure, they hope they won't need to use them, even on the Unfettered, but they value all life — they would try to apprehend rather than blow to pieces, given the chance. I wouldn't be surprised if peaceful squirrels practiced some martial arts (with emphasis on non-lethal disabling moves or pain compliance techniques) as well. And while these rifles would most likely be used in the conflict with the Unfettered, there's no reason to think that they are lacking in security elsewhere. Anyone (including humans and Hai themselves) is a potential threat to peace, it wouldn't last had they not be able to enforce it.

2. **We probably shouldn't worry about making laser rifles obsolete.** This would apply only to players who have found the Hai anyway, and then we could say that their reactors and batteries and other tech make human ones obsolete, and so on. This wasn't an issue before, and it isn't likely to become an issue now, it is part of the game's progression.

**Comment by Nescio0** [comment]
To clarify, I actually like the looks of this new weapon. I'm just unconvinced making it widely available is a good idea. It would be less of an issue if it was only added to Unfettered variants and not to any outfitters or default ships.

> then we could say that their reactors and batteries and other tech make human ones obsolete, and so on

Hai anti-missile turrets, atomic engines, batteries, and reactors are slightly smaller and more efficient than their human equivalents, but also more expensive and slightly weaker. This rifle is however clearly stronger while the Hai are a pacifist species. Perhaps its stats could be lowered to slightly below the human's rifle, to e.g. +0.5 attack and +0.75 defence (human rifle has 0.6/0.8)?

**Comment by Brick63** [comment]
That would not make sense as the hai are more advanced than humanity

**Comment by Nescio0** [comment]
> That would not make sense as the hai are more advanced than humanity

Being generally more advanced doesn't mean everything has to be more powerful, nor is it without precedent. E.g.:
```
                           thrust
"Bufaer" Atomic Thruster : 432,360
A860 Atomic Thruster     : 502,920
Tyrant Plasma Thruster   : 469,800
X5700 Ion Thruster       : 293,400
```

Also, our modern combat knifes are flimsier than Medieval blades, even though humanity is generally more advanced than it was a millenium ago.

**Comment by Brick63** [comment]
Well melee weapons have fallen out of use as the main form of combat and people no longer wear full suits of metal to go get groceries, and the precedent is that they are more advanced than humanity and it shows in everything

**Comment by Brick63** [comment]
Also their thusters are closer to the second largest human atomic in size iirc

**Comment by Nescio0** [comment]
> they are more advanced than humanity and it shows in everything

Including in being peaceful and not walking around with combat weapons.

**Comment by Brick63** [comment]
If they are so peaceful then why are all their non Frontline ships armed

**Comment by Brick63** [comment]
They use force as a last resort

**Comment by Brick63** [comment]
Neither does the Navy yet they're armed to the teeth

**Comment by Tadrix** [comment]
This outfit probably should be kept out of Hai Basics outfitters (which are farthest away from the Unfettered) — I imagine back on Allhome they wouldn't want to greet humans with "Hi, we are a peaceful race, and, by the way, take a look at our latest in hand to hand technology, these rifles are a blast to use!" And the Unfettered probably wouldn't be selling these to outsiders at all. Closer to the conflict (Intermediate and Advanced outfitters) they are a fair game: "Protect yourselves in case you get boarded. And please, don't zap them to death."

**Comment by endless-sky** [comment]
Given that frag grenades will still be the best legal way to capture ships, I don't see a problem with a Hai hand to hand weapon being better across the board than the human equivalent. But it might make sense to have the stats be more like 0.7 / 1.0, i.e. just a small improvement over the human tech.

Also, instead of making it so literally every Hai ship is carrying nearly a full complement of hand to hand weapons, it'd make sense for some Hai ships to carry none at all (especially the stock ships that you can purchase in a shipyard) and the rest to have anywhere from just a few guns up to a full complement.

**Comment by Brick63** [comment]
i see where you're coming from

**Comment by endless-sky** [comment]
By "full complement" I meant the required crew, not the number of bunks. In general it doesn't make sense for the number of hand-to-hand weapons on a ship to be greater than the number of required crew members.

Of course, on ships that are only used by the Unfettered, maybe a few could have more guns than they could use as a sort of subtle reflection of the Unfettered personality. After all, even among humans it's not unheard of in certain subcultures for a household to contain more guns than people. :)

**Comment by tehhowch** [comment]
several times more, even O:)

File: data/hai ships.txt [hunk 5]
```diff
@@ -36,7 +36,6 @@ ship "Aphid"
 		"Hai Fissure Batteries"
 		"Hai Corundum Regenerator"
 		"Hai Williwaw Cooling"
-
```
Comment (tehhowch) [inline data/hai ships.txt]: put this back :)

**Comment by Brick63** [comment]
i think that should be it
